.ci/generate_redirects.jl

```diff
@@ -89,13 +95,60 @@ function create_redirect_page(; name, path)
     end
 end
 
+function create_404_page()
+    open(joinpath("webroot", "404.html"), "w") do io
+        write(io, """
+        <!DOCTYPE html>
+        <html lang="en">
+        <head>
+            <meta charset="UTF-8">
+            <meta name="viewport" content="width=device-width, initial-scale=1.0">
+            <title>Page not found</title>
+            $(style_block("#ffcbc8", "#CB3C33"))
+        </head>
+        <body>
+            <div class="centered-div">
+                <p>No page exists here.<br><br>Redirection pages for registered Julia packages follow the format packages/redirect_to_repo/SomePackage.</p>
+            </div>
+
+            <script>
+            // Get the current URL path
+            const urlPath = window.location.pathname;
+
+            // Define the regex pattern to match the URL structure
+            const pattern = /\\/packages\\/redirect_to_repo\\/([^\\/]+?)(\\.html)?\$/;
+
+            // Check if the URL matches the pattern
+            const match = urlPath.match(pattern);
+
+            if (match) {
+                const packageName = match[1]; // Extract the package name
+                const messageElement = document.querySelector(".centered-div p");
+                // Update the paragraph text
+                messageElement.innerHTML = `\
+                    There is no package \${packageName} registered in the Julia General Registry.\
+                    <br><br>\
+                    Would you like to try searching <a href="https://github.com/search?q=\${packageName}.jl&type=repositories">Github</a>, \
+                    <a href="https://about.gitlab.com/search/?searchText=\${packageName}.jl">Gitlab</a>, \
+                    <a href="https://www.google.com/search?q=\${packageName}.jl">Google</a>, \
+                    or <a href="https://duckduckgo.com/?t=h_&q=\${packageName}.jl">DuckDuckGo</a> for it?`;
```

Comment:
What does the `t=h_` do?

Reply:
Looks like it's used by partners/affiliates of DDG? https://duckduckgo.com/duckduckgo-help-pages/privacy/t/

Seems like we should remove the `t=` parameter?

Reply:
```suggestion
                    or <a href="https://duckduckgo.com/?q=\${packageName}.jl">DuckDuckGo</a> for it?`;
```

Reply:
ah sorry about that. The URL came from me copy-pasting from Firefox.